src/net/udp.rs

```diff
@@ -49,16 +52,23 @@ impl UdpSocket {
     /// Sends data on the socket to the given address. On success, returns the number of bytes written.
     pub fn send(&mut self, packet: Packet) -> NetworkResult<usize> {
         let (addr, mut packet_data) = self.state.pre_process_packet(packet, &self.config)?;
-
         let mut bytes_sent = 0;
-
-        for payload in packet_data.parts() {
-            bytes_sent += self
-                .socket
-                .send_to(&payload, addr)
-                .map_err(|io| NetworkError::from(NetworkErrorKind::IOError { inner: io }))?;
+        if let Some(link_conditioner) = &self.link_conditioner {
+            if link_conditioner.should_send() {
+                for payload in packet_data.parts() {
```

Comment:
might be good to move this out to its own function because its repeated below.

Reply:
#78